{
  try {
    return task.call();
  }
 catch (  Throwable t) {
    t.printStackTrace();
    if (t instanceof Exception)     throw (Exception)t;
 else     throw new RuntimeException(t);
  }
}
